MantidQt/CustomInterfaces/src/SANSRunWindow.cpp

```diff
@@ -1702,18 +1700,21 @@ void SANSRunWindow::setGeometryDetails() {
     return;
   }
 
-  const auto &dets = monitorWs->getSpectrum(monitorWsIndex).getDetectorIDs();
-  if (dets.empty())
+  const auto &monitorDetectorIDs =
+      monitorWs->getSpectrum(monitorWsIndex).getDetectorIDs();
+  if (monitorDetectorIDs.empty())
     return;
 
   double dist_mm(0.0);
   QString colour("black");
-  try {
-    Mantid::Geometry::IDetector_const_sptr detector =
-        instr->getDetector(*dets.begin());
 
-    double unit_conv(1000.);
-    dist_mm = detector->getDistance(*source) * unit_conv;
+  const auto &spectrumInfo = sampleWs->spectrumInfo();
+
+  try {
+    const auto &detector = spectrumInfo.detector(*monitorDetectorIDs.begin());
+    const double unit_conv(1000.);
+    const auto &source = sampleWs->getInstrument()->getSource();
+    dist_mm = detector.getDistance(*source) * unit_conv;
```

Comment:
@SimonHeybrock I'm a little bit confused - If  "MatrixWorkspace should use SpectrumInfo" in the proposed changes you agree using `DetectorInfo` is correct however `sampleWS` is a Matrix Workspace and not an instrument?